src/usr/local/www/diag_dns.php

```diff
@@ -84,12 +84,33 @@
 }
 
 function resolve_host_addresses($host) {	
-	$host_esc = escapeshellarg($host);
 	$resolved = array();
-	exec("/usr/bin/drill {$host_esc} A | /usr/bin/grep {$host_esc} | grep '\tA\t\|\tCNAME\t' | /usr/bin/grep -v ';' | /usr/bin/awk '{ print $5 }'", $resolved);
-	$resolved_v6 = array();
-	exec("/usr/bin/drill {$host_esc} AAAA | /usr/bin/grep {$host_esc} | grep '\tAAAA\t\|\tCNAME\t' | /usr/bin/grep -v ';' | /usr/bin/awk '{ print $5 }'", $resolved_v6);
-	return array_unique(array_merge($resolved, $resolved_v6));
+
+	$errreporting = error_reporting();
+	error_reporting($errreporting & ~E_WARNING);// dns_get_record throws a warning if nothing is resolved..
+
+	$dnsresult_cname = dns_get_record($host, DNS_CNAME);
+	$dnsresult_v4 = dns_get_record($host, DNS_A);
+	$dnsresult_v6 = dns_get_record($host, DNS_AAAA);	
```

Comment:
Nevermind, it will return false if one of these items are not available. I did some tests and looks like the best way of doing that is running a single call using DNS_ALL, it'll return all CNAME, A and AAAA records.

It's better to use a single call because of timeout in case of problems, it will be reduced by 2/3